src/karthik/group-people-data/groupPeopleData.ts

```diff
@@ -0,0 +1,44 @@
+interface People {
+  _id: string;
+  name: string;
+  gender: string;
+  email: string;
+  phone: string;
+  city: string;
+}
+
+interface GroupOfPeopleByCity {
+  count: number;
+  city: string;
+  people: Array<People>;
+}
+
+const groupPeopleByCity = (
+  arrayOfObject: Array<People>
+): Array<GroupOfPeopleByCity> => {
+  if (arrayOfObject === []) return [];
```

Comment:
For catching the blank array, prefer different ways by type checking or using length property than this, as `[] === []` return `false` in JS.
How about some thing like this?
```
if(!arrayOfObject || arrayOfObject.length < 1) return false;
```
```
if(!arrayOfObject || !arrayOfObject.length) return false;
```
```
if(typeof arrayOfObject !== 'object' || !arrayOfObject.length) return false;
```